src/Scaler/Services/CosmosDbMetricProvider.cs

```diff
@@ -3,47 +3,73 @@
 using System.Net;
 using System.Net.Http;
 using System.Threading.Tasks;
+using Azure.Identity;
+using Microsoft.AspNetCore.Http;
 using Microsoft.Azure.Cosmos;
 using Microsoft.Extensions.Logging;
-
+using Microsoft.Identity.Client.Platforms.Features.DesktopOs.Kerberos;
+using Container = Microsoft.Azure.Cosmos.Container;
+using System.Diagnostics.Metrics;
+using OpenTelemetry;
+using OpenTelemetry.Metrics;
+using Microsoft.Identity.Client;
 namespace Keda.CosmosDb.Scaler
 {
     internal sealed class CosmosDbMetricProvider : ICosmosDbMetricProvider
     {
         private readonly CosmosDbFactory _factory;
         private readonly ILogger<CosmosDbMetricProvider> _logger;
 
+		static Meter s_meter = new Meter("OrderProcessor.CFStore", "1.0.0");
+        private static Counter<long> s_CFRecordsReceived = s_meter.CreateCounter<long>("RecordsReceived");
+        private static Counter<int> s_CFProcessorCount = s_meter.CreateCounter<int>("ProcessorCount");
         public CosmosDbMetricProvider(CosmosDbFactory factory, ILogger<CosmosDbMetricProvider> logger)
         {
             _factory = factory ?? throw new ArgumentNullException(nameof(factory));
             _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+
+            using MeterProvider meterProvider = Sdk.CreateMeterProviderBuilder()
+               .AddMeter("OrderProcessor.CFStore")               
+               .Build();
         }
 
         public async Task<long> GetPartitionCountAsync(ScalerMetadata scalerMetadata)
         {
             try
             {
-                Container leaseContainer = _factory
-                    .GetCosmosClient(scalerMetadata.LeaseConnection)
-                    .GetContainer(scalerMetadata.LeaseDatabaseId, scalerMetadata.LeaseContainerId);
 
-                ChangeFeedEstimator estimator = _factory
-                    .GetCosmosClient(scalerMetadata.Connection)
-                    .GetContainer(scalerMetadata.DatabaseId, scalerMetadata.ContainerId)
-                    .GetChangeFeedEstimator(scalerMetadata.ProcessorName, leaseContainer);
+                var credential = new DefaultAzureCredential();
+                var cosmosClient = new CosmosClient(scalerMetadata.Endpoint, credential);
```

Comment:
Use `GetCosmosClient` to have a single connection if both endpoints happen to be same.

Reply:
OK